solutions/fully-configurable/main.tf

```diff
@@ -230,4 +230,27 @@ module "ocp_base" {
   use_private_endpoint                  = var.use_private_endpoint
   verify_worker_network_readiness       = var.verify_worker_network_readiness
   worker_pools_taints                   = var.worker_pools_taints
+  enable_secrets_manager_integration    = var.enable_secrets_manager_integration
+  existing_secrets_manager_instance_crn = var.existing_secrets_manager_instance_crn
+  secrets_manager_secret_group_id       = var.secrets_manager_secret_group_id != null ? var.secrets_manager_secret_group_id : (var.enable_secrets_manager_integration ? module.secret_group[0].secret_group_id : null)
+}
+
+module "existing_secrets_manager_instance_parser" {
+  count   = var.enable_secrets_manager_integration ? 1 : 0
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = var.existing_secrets_manager_instance_crn
+}
+
+module "secret_group" {
+  count                    = var.enable_secrets_manager_integration && var.secrets_manager_secret_group_id == null ? 1 : 0
```

Comment:
Your going to have to add a new provider alias in the provider.tf file for working with secrets manager resources incase the instance is in a different region to the OCP cluster. We already do this for EN and KMS in the provider.tf file. Just copy same approach